### PR TITLE
Set up Jest to be able to test RN UI components

### PR DESCRIPTION
**Pull request by ferrannp**
With this we can test UI like `TabNavigator` or `DrawerNavigator`. We can also use [Jest snapshot](https://facebook.github.io/jest/docs/tutorial-react-native.html#snapshot-test).

It required some Jest extra configuration.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,9 +70,14 @@
     "react-native-tab-view": "^0.0.54"
   },
   "jest": {
-    "testPathDirs": [
```
Comment (ferrannp) [inline package.json]: This config had issues with RN, see https://github.com/facebook/jest/issues/2670.

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,6 @@ yarn-error.log
 
 # Exponent
 .exponent
+
+# Jest
+coverage
```
Comment (ferrannp) [inline .gitignore]: Jest will generate a coverage folder if you run `npm test -- --coverage`.

File: jest-setup.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* @flow */
+
+// See https://github.com/facebook/jest/issues/2208
+jest.mock('Linking', () => {
```
Comment (satya164) [inline jest-setup.js]: Can you add the mock for ScrollView too?

File: package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
     "start": "node ./node_modules/react-native/local-cli/cli.js start --config ./rn-cli.config.js",
     "run-playground-ios": "cd examples/NavigationPlayground && react-native run-ios",
     "run-playground-android": "cd examples/NavigationPlayground && react-native run-android",
-    "test": "BABEL_ENV=development jest",
```
Comment (ferrannp) [inline package.json]: This was not necessary. Actually, with this line, the `test` env in `.babelrc` was totally ignored.

File: jest-setup.js (hunk 3 above)
Comment (ferrannp) [inline jest-setup.js]: You got it ;)

File: src/navigators/__tests__/__snapshots__/DrawerNavigator-test.js.snap [hunk 5]
```diff
@@ -0,0 +1,167 @@
+exports[`DrawerNavigator renders a DrawerNavigator with correct options 1`] = `
```
Comment (satya164) [inline src/navigators/__tests__/__snapshots__/DrawerNavigator-test.js.snap]: Remove this, we can snapshot views later.

File: src/navigators/__tests__/DrawerNavigator-test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+/* @flow */
+
+import 'react-native';
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import DrawerNavigator from '../DrawerNavigator';
+
+const _getScreen = (label: string) => {
```
Comment (torabian) [inline src/navigators/__tests__/DrawerNavigator-test.js]: What is this `label: string` sounds like TypeScript, but in JavaScript. Is it flow?

**Pull request merged**